ANTLR/src/Compiler/Main.java

```diff
@@ -22,33 +23,47 @@ public static void main(String[] args) throws IOException {
             throw new IOException("Filename must have the suffix: '.buff'.");
 
-        CharStream stream = CharStreams.fromFileName(args[0]);
+        try{
+            CharStream stream = CharStreams.fromFileName(args[0]);
 
-        // Syntax analysis
-        LangLexer lexer = new LangLexer(stream);
-        CommonTokenStream tokens = new CommonTokenStream(lexer);
-        LangParser parser = new LangParser(tokens);
-        ParseTree tree = parser.prog();
+            //Error handling
+            UnderlineErrorListener errorListener = new UnderlineErrorListener();
 
-        // Symbol table generation
-        ParseTreeWalker walker = new ParseTreeWalker();
-        SymbolTableGeneratorListener symbolTable = new SymbolTableGeneratorListener();
-        walker.walk(symbolTable, tree);
+            // Syntax analysis
+            LangLexer lexer = new LangLexer(stream);
+            lexer.removeErrorListeners();
+            lexer.addErrorListener(errorListener);
+            CommonTokenStream tokens = new CommonTokenStream(lexer);
+            LangParser parser = new LangParser(tokens);
+            parser.removeErrorListeners();
+            parser.addErrorListener(errorListener);
+            ParseTree tree = parser.prog();
 
+            // Symbol table generation
+            ParseTreeWalker walker = new ParseTreeWalker();
+            SymbolTableGeneratorListener symbolTable = new SymbolTableGeneratorListener(errorListener);
+            walker.walk(symbolTable, tree);
 
-        // Contextual analysis
-        ReferenceCheckerListener referenceChecker = new ReferenceCheckerListener(symbolTable.globalScope, symbolTable.scopes);
-        walker.walk(referenceChecker, tree);
 
-        TypeCheckerVisitor typeChecker = new TypeCheckerVisitor(symbolTable.globalScope, symbolTable.scopes);
-        typeChecker.visit(tree);
+            // Contextual analysis
+            ReferenceCheckerListener referenceChecker =
+                    new ReferenceCheckerListener(symbolTable.globalScope, symbolTable.scopes, errorListener);
+            walker.walk(referenceChecker, tree);
 
+            TypeCheckerVisitor typeChecker =
+                    new TypeCheckerVisitor(symbolTable.globalScope, symbolTable.scopes, errorListener);
+            typeChecker.visit(tree);
 
-        // Code generation
-        JavaScriptCodeGenerationVisitor codeGenerator = new JavaScriptCodeGenerationVisitor();
-        String targetCode = codeGenerator.visit(tree);
 
-        OutputFile output = new OutputFile(targetCode, args);
-        output.execute();
+            // Code generation
+            JavaScriptCodeGenerationVisitor codeGenerator = new JavaScriptCodeGenerationVisitor();
+            String targetCode = codeGenerator.visit(tree);
+
+            OutputFile output = new OutputFile(targetCode, args);
+            output.execute();
+        }catch (Exception e){
+            //Do not know what makes sense to do here if anything, so empty for now.
```

Comment:
When will this happen?

Reply:
Whenever ThrowError is called in the errorListener. / Whenever an error is occurred.